apps/documents/serializers.py

```diff
@@ -6,38 +6,32 @@
 
 from enumfields.drf import EnumSupportSerializerMixin
 from rest_framework_json_api import serializers
+from rest_framework import exceptions
 from influxdb_metrics.loader import log_metric
 
 from apps.utils import UpperEnumField
-from .models import Document, DocumentType, FileType, UploadStatus, IMAGE_TYPES
+from .models import Document, CsvDocument, DocumentType, FileType, CsvFileType, UploadStatus, ProcessingStatus
 from .rpc import DocumentRPCClient
 
 LOG = logging.getLogger('transmission')
 
 
 class DocumentSerializer(EnumSupportSerializerMixin, serializers.ModelSerializer):
-    document_type = UpperEnumField(DocumentType, lenient=True, read_only=True, ints_as_names=True)
-    file_type = UpperEnumField(FileType, lenient=True, read_only=True, ints_as_names=True)
-    upload_status = UpperEnumField(UploadStatus, lenient=True, ints_as_names=True)
-    presigned_s3 = serializers.SerializerMethodField()
-
-    class Meta:
-        model = Document
-        if settings.PROFILES_ENABLED:
-            exclude = ('owner_id', 'shipment',)
-        else:
-            exclude = ('shipment',)
 
     def get_presigned_s3(self, obj):
-        file_extension = obj.file_type.name.lower()
-
-        if file_extension in IMAGE_TYPES:
-            content_type = f"image/{file_extension}"
+        if obj.__class__.__name__ == 'Document':
```

Comment:
I do not understand what you mean by that? This base serializer class handle two classes: `Document` and `CsvDocment` , that line of code is to differentiate them.

Reply:
well, the if is checking for a document then defaulting to a CSV rather than the other way around. I was just thinking that in case we add more document types, so we would have csv/??/document it would make sense to default with the else to document

Reply:
I see now. I don't think it really matters also because `Document` will be called more often than `CsvDocument` and the `else` statement will be called just a few times.